apps/studio/components/interfaces/Auth/Policies/AIPolicyEditorPanel/AIPolicyChat.tsx

```diff
@@ -0,0 +1,163 @@
+          icon={<AiIcon className="[&>div>div]:border-white" />}
+          postedBy="Assistant"
+          message={`Hi${
+            name ? ' ' + name : ''
+          }, how can I help you? I'm powered by AI, so surprises and mistakes are possible.
+        Make sure to verify any generated code or suggestions, and share feedback so that we can
+        learn and improve.`}
+        />
+
+        {sorted.map((m) => {
+          const content = m.content[0]
+          if (content && content.type !== 'text') {
+            return null
+          }
+
+          return (
+            <Message
+              key={m.id}
+              icon={
+                m.role === 'assistant' ? (
+                  <AiIcon className="[&>div>div]:border-white" />
+                ) : (
+                  <div className="relative border shadow-lg w-8 h-8 rounded-full overflow-hidden">
+                    <Image
+                      src={`https://github.com/${profile?.username}.png` || ''}
+                      width={30}
+                      height={30}
+                      alt="avatar"
+                      className="relative"
+                    />
+                  </div>
+                )
+              }
+              postedBy={m.role === 'assistant' ? 'Assistant' : name ? name : 'You'}
+              postedAt={m.created_at}
+              message={content.text.value}
+              onDiff={onDiff}
+              isDebug={(m.metadata as any).type === 'debug'}
+            />
+          )
+        })}
+
+        {pendingReply && (
+          <Message
+            icon={<AiIcon className="[&>div>div]:border-white" />}
+            postedBy={'Assistant'}
+            message={'Thinking...'}
+          />
+        )}
+
+        <div ref={bottomRef} className="h-1" />
+      </div>
+      <Form_Shadcn_ {...form}>
+        <form
+          id="rls-chat"
+          className="sticky p-5 flex-0 border-t"
+          onSubmit={form.handleSubmit((data: z.infer<typeof FormSchema>) => {
+            setIsLoading(true)
+            onSubmit(data.chat)
+          })}
+        >
+          <FormField_Shadcn_
+            control={form.control}
+            name="chat"
+            render={({ field }) => (
+              <FormItem_Shadcn_>
+                <FormControl_Shadcn_>
+                  <div className="relative">
+                    <AiIcon className="absolute top-2 left-3 [&>div>div]:border-white" />
+                    <Input_Shadcn_
+                      {...field}
+                      autoComplete="off"
+                      disabled={isLoading}
+                      className={`bg-black rounded-full pl-10 ${isLoading ? 'pr-10' : ''}`}
+                      placeholder="Ask for some changes to your policy"
+                    />
+                    {isLoading && <Loader2 className="absolute top-2 right-3 animate-spin" />}
+                  </div>
+                </FormControl_Shadcn_>
+              </FormItem_Shadcn_>
```

Comment:
confirm close side panel
https://share.cleanshot.com/B3LHy3Zh

Reply:
also check if theres an ongoing conversation, if yes show confirm close side panel
confirm close side panel should also indicate that any ongoing conversation with assistant will be lost